### PR TITLE
Issue 7: Delete Image

### PR DESCRIPTION
**Pull request by chen-daniel**
Closes Issue 7

### CODE REVIEW

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -6,7 +6,7 @@
   </tr>
   <% @images.each do |i| %>
     <tr>
-      <td><%= image_tag(i.url, class: 'home-image') %></td>
+      <td><a href="<%= image_path(i.id) %>"><%= image_tag(i.url, class: 'home-image') %></a></td>
```
Comment (raj-al-ghul) [inline app/views/images/index.html.erb]: lets use the helpers instead of `a` directly

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -131,4 +131,20 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    end
+    assert_redirected_to root_path
+  end
+
+  test 'destroy should fail to remove an image not in the db' do
+    image = Image.create!(url: 'https://www.w3schools.com/w3css/img_lights.jpg')
+    assert_no_difference 'Image.count' do
+      delete image_path(image.id + 1)
+    end
+    assert_response :unprocessable_entity
```
Comment (raj-al-ghul) [inline test/controllers/images_controller_test.rb]: is there anything else we do in this case that we can test?

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -25,6 +25,16 @@ def show
     @image = Image.find_by(id: params[:id])
   end
 
+  def destroy
+    image = Image.find_by(id: params[:id])
+    if image.nil?
+      redirect_to root_path, status: :unprocessable_entity
```
Comment (raj-al-ghul) [inline app/controllers/images_controller.rb]: lets discuss if there is anything else we could want to do in this case

Comment (chen-daniel) [reply app/controllers/images_controller.rb]: Added flash messages for error and successful completion

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -131,4 +131,22 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
     assert_select 'img', false
   end
+
+  test 'destroy should remove an image in the db' do
+    image = Image.create!(url: 'https://www.w3schools.com/w3css/img_lights.jpg')
+    assert_difference 'Image.count', -1 do
+      delete image_path(image.id)
+    end
+    assert_redirected_to root_path
+    assert_equal 'Image successfully deleted', flash[:notice]
```
Comment (raj-al-ghul) [inline test/controllers/images_controller_test.rb]: do we check if it shows up in the UI?

File: app/views/images/new.html.erb [hunk 5]
```diff
@@ -1,19 +1,6 @@
 <a href="<%= root_path %>">Home</a>
 <h1>Add an image URL</h1>
 <%= simple_form_for @image do |f| %>
-  <% if @image.errors.any? %>
```
Comment (raj-al-ghul) [inline app/views/images/new.html.erb]: were there no tests that had to be updated?

File: app/views/images/index.html.erb [hunk 6]
```diff
@@ -1,3 +1,9 @@
+<% if flash[:notice] %>
```
Comment (raj-al-ghul) [inline app/views/images/index.html.erb]: is there a way to have one alert that will cover both cases?

**Comment by chen-daniel** [comment]
[Issue 7: Delete Image](https://trello.com/c/oWHyXr29/76-issue-7-delete-image)

**Pull request merged**